packages/@glimmer/manager/lib/public/modifier.ts

```diff
@@ -99,41 +98,44 @@ export class CustomModifierManager<O extends Owner, ModifierInstance>
     let delegate = this.getDelegateFor(owner);
 
     let args = argsProxyFor(capturedArgs, 'modifier');
-    let instance: ModifierInstance = delegate.createModifier(definition, args);
+    let modifier: ModifierStateBucket = delegate.createModifier(definition, args);
 
     let tag = createUpdatableTag();
-    let state: CustomModifierState<ModifierInstance>;
+    let state: CustomModifierState<ModifierStateBucket>;
 
     state = {
       tag,
       element,
       delegate,
       args,
-      modifier: instance,
+      modifier,
     };
 
-    registerDestructor(state, () => delegate.destroyModifier(instance, args));
+    registerDestructor(state, () => delegate.destroyModifier(modifier, args));
 
     return state;
   }
 
   getDebugName(definition: object) {
-    if (typeof definition === 'function') {
-      return definition.name || definition.toString();
-    } else {
-      return '<unknown>';
+    const delegate = this.factory.prototype;
+    if (typeof delegate?.getDebugName === 'function') {
```

Comment:
yeah, I tried that, the problem is in https://github.com/glimmerjs/glimmer-vm/blob/10eae7429b702b1e7f5434b91802d5767ff7ad9a/packages/%40glimmer/runtime/lib/environment.ts#L68
there is no direct way to access the owner, maybe more changes would be requried?

Reply:
oof, sorry this has been *rough*. It's the code equivalent of random bureaucracy. Notionally we have the information, just not consistently threaded/easily accessible.

Since we narrowed this problem down to these private APIs (I think, this class is private, right?), I don't really care that much about the API surface.

I think the API surface of what we call the "delegate" here is important (that's what people actually implement right?), and I think we got that down pretty good.

As long as we don't affect the public API, we can do anything (that is type safe, etc) for this wrapper class. Maybe we keep `getDebugName(owner, definition)` and `getDebugNameFromInstance(instance)` in this internal class and call it a day for now. For reference: this is the kind of place we may want to use the former, which we _do_ have the owner: https://github.com/glimmerjs/glimmer-vm/blob/34888e9b0de56475c1c9fab88c4007816dd1bf44/packages/%40glimmer/runtime/lib/compiled/opcodes/dom.ts#L147-L153

For what its worth I think a lot of this up is overdue for a cleanup/rethink pass, especially for the manages stuff. The pure form of the idea is good, but we made some decisions along the way that incidentally complicated things a fair bit.